src/posts/lucky_1.2.0.cr

```diff
@@ -0,0 +1,237 @@
+
+    Finally, `String` also received a few more Criteria related to Full-Text searching.
+
+    Convert a `String` column `to_tsquery` or `to_tsvector` by using those methods names in
+    conjunction with the new `match` method.
+
+    ```crystal
+    # SELECT * FROM profiles WHERE TO_TSVECTOR(description) @@ TO_TSQUERY('Lucky')
+    ProfileQuery.new.description.to_tsvector.match("Lucky")
+    ```
+
+    ### New migration methods
+
+    Migrations received a few updates. You can now run your `create` and `alter` conditionally
+    with `if_not_exists` and `if_exists` respectively.
+
+    ```crystal
+    # Only create the table if it does not already exist
+    create table_for(Server), if_not_exists: true do
+      add hostname : String
+    end
+    ```
+
+    You can now ignore the index created on `add_belongs_to` by specifying `index: false`.
+
+    ```crystal
+    alter table_for(Reaction), if_exists: true do
+      add_belongs_to user : User?, index: false
+      add_belongs_to friend : User?, index: false
+    end
+    ```
+
+    This will allow you to create a custom unique index utilizing multiple columns instead of separate indexes on each column.
+
+    We've also added the `create_sequence` and `drop_sequence` helpers.
+
+    ```crystal
+    def migrate
+      create_sequence table_for(Account), :number
+    end
+
+    def rollback
+      drop_sequence table_for(Account), :number
+    end
+    ```
+
+    ### The IGNORE constant
+
+    In Avram, the attributes defined on operations are a union of the type you specified, and `Avram::Nothing`.
+    The `Avram::Nothing` type is used as a way to differentiate between you intending to update a column to `nil`
+    by passing in a `nil` value versus a `nil` value existing in params but with no intention of updating the column.
+
+    Previously, each attribute would define a default value of `Avram::Nothing.new`. The larger your application grew,
+    the more instances of `Avram::Nothing` you would have. These all have now been replaced with a single constant `IGNORE`.
+
+    Use the `IGNORE` constant when you want to fallback to ignoring a column in a SaveOperation.
+
+    ```crystal
+    # If `new_email.presence` is nil, then just ignore this update
+    UpdateUserEmail.update!(
+      current_user,
+      email: new_email.presence || IGNORE
+    )
+    ```
+
+    ### Email attachments
+
+    [Carbon](https://github.com/luckyframework/carbon/pull/88) received a nice update with the addition of
+    attachment support.
+
+    ```crystal
+    class BirthdayEmail < BaseEmail
+      def initiailize(@recipient : Carbon::Emailable)
+        @image = File.open("birthday_cake.png")
+      end
+
+      subject "Happy Birthday"
+      to @recipient
+      templates html
+      attachment cake
+
+      def cake
+        {
+          io: @image,
+          file_name: "birthday_cake.png",
+          mime_type: "image/png"
+        }
+      end
+
+      def after_send(result)
+        @image.close
```

Comment:
Do devs need to remember to close the file? If so, that would be a good thing to mention, even in a comment:

```suggestion
        # remember to close your open files!
        @image.close
```

Reply:
I didn't implement the attachments, so I wasn't sure how this worked, to be honest. They take an `IO` object, but `File.read()` returns a `String`, so I just did this as a shot in the dark :joy: I guess I could swap to using `IO.copy` then the file.close is irrelevant. Would that be better for this example?

Reply:
I'm now using IO.copy in the example. It's probably better anyway because I think if the email fails to send, the `after_send` isn't called :thinking: